include/deal.II/lac/read_write_vector.h

```diff
@@ -0,0 +1,715 @@
+     */
+    template <typename Number2>
+    ReadWriteVector<Number> &
+    operator= (const ReadWriteVector<Number2> &in_vector);
+
+#ifdef DEAL_II_WITH_PETSC
+    /**
+     * Imports all the elements present in the vector's IndexSet from the input
+     * vector @p petsc_vec.
+     */
+    ReadWriteVector<Number> &
+    operator= (const PETScWrappers::MPI::Vector &petsc_vec);
+#endif
+
+#ifdef DEAL_II_WITH_TRILINOS
+    /**
+     * Imports all the elements present in the vector's IndexSet from the input
+     * vector @p trilinos_vec.
+     */
+    ReadWriteVector<Number> &
+    operator= (const TrilinosWrappers::MPI::Vector &trilinos_vec);
+#endif
+
+    /**
+     * Sets all elements of the vector to the scalar @p s. This is operation is
+     * allowed if @p is equal to zero.
+     */
+    ReadWriteVector<Number> &operator = (const Number s);
+
+    /**
+     * Returns the size of the vector, equal to number of elements in
+     * the vector.
+     */
+    size_type size () const;
+
+    /**
+     * Return the IndexSet that stores the indices of the elements stored.
+     */
+    const IndexSet &get_stored_elements () const;
+
+    /**
+     * Make the @p ReadWriteVector class a bit like the <tt>vector<></tt> class of
+     * the C++ standard library by returning iterators to the start and end
+     * of the <i>locally stored</i> elements of this vector.
+     *
+     * It holds that end() - begin() == size().
+     */
+    iterator begin ();
+
+    /**
+     * Returns constant iterator to the start of the locally stored elements
+     * of the vector.
+     */
+    const_iterator begin () const;
+
+    /**
+     * Returns an iterator pointing to the element past the end of the array
+     * of locally stored entries.
+     */
+    iterator end ();
+
+    /**
+     * Returns a constant iterator pointing to the element past the end of
+     * the array of the locally stored entries.
+     */
+    const_iterator end () const;
+    //@}
+
+
+    /**
+     * @name 2: Data-Access
+     */
+    //@{
+
+    /**
+     * Read access to the data in the position corresponding to @p
+     * global_index.
+     */
+    Number operator () (const size_type global_index) const;
+
+    /**
+     * Read and write access to the data in the position corresponding to @p
+     * global_index.
+     */
+    Number &operator () (const size_type global_index);
+
+    /**
+     * Read access to the data in the position corresponding to @p
+     * global_index.
+     *
+     * This function does the same thing as operator().
+     */
+    Number operator [] (const size_type global_index) const;
+
+    /**
+     * Read and write access to the data in the position corresponding to @p
+     * global_index.
+     *
+     * This function does the same thing as operator().
+     */
+    Number &operator [] (const size_type global_index);
+
+    /**
+     * A collective get operation: instead of getting individual elements of
```

Comment:
what do you mean by "collective operation", this has nothing to do with MPI collectives (the first thing that came to my mind).